src/dex/solidly/solidly-e2e.test.ts

```diff
@@ -0,0 +1,613 @@
+        describe(`Stable`, () => {
+          it('USDC -> USDT', async () => {
+            await testE2E(
+              tokens.USDC,
+              tokens.FUSDT,
+              holders.USDC,
+              '100000',
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.simpleSwap,
+              network,
+              provider,
+            );
+          });
+        });
+      });
+      describe(`multiSwap`, () => {
+        describe(`Volatile`, () => {
+          it('FTM -> USDC', async () => {
+            await testE2E(
+              tokens.FTM,
+              tokens.USDC,
+              holders.FTM,
+              '1000000000000000000',
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.multiSwap,
+              network,
+              provider,
+            );
+          });
+          it('FTM -> SPIRIT', async () => {
+            await testE2E(
+              tokens.FTM,
+              tokens.SPIRIT,
+              holders.FTM,
+              '11000000000000000000',
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.multiSwap,
+              network,
+              provider,
+            );
+          });
+          it('SPIRIT -> FTM', async () => {
+            await testE2E(
+              tokens.SPIRIT,
+              tokens.FTM,
+              holders.SPIRIT,
+              '11000000000000000000',
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.multiSwap,
+              network,
+              provider,
+            );
+          });
+          it('SPIRIT -> WFTM', async () => {
+            await testE2E(
+              tokens.SPIRIT,
+              tokens.WFTM,
+              holders.SPIRIT,
+              '1000000000000000000',
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.multiSwap,
+              network,
+              provider,
+            );
+          });
+        });
+        describe(`Stable`, () => {
+          it('USDC -> USDT', async () => {
+            await testE2E(
+              tokens.USDC,
+              tokens.FUSDT,
+              holders.USDC,
+              '100000',
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.multiSwap,
+              network,
+              provider,
+            );
+          });
+        });
+      });
+    });
+  });
+
+  describe('Polygon', () => {
+    const network = Network.POLYGON;
+    const tokens = Tokens[network];
+    const holders = Holders[network];
+    const provider = new StaticJsonRpcProvider(
+      generateConfig(network).privateHttpProvider,
+      network,
+    );
+
+    describe('Dystopia', () => {
+      const dexKey = 'Dystopia';
+      const usdAmount = '1000000';
+
+      describe('Dystopia UniswapV2 Pools', () => {
+        const maticAmount = '1000000000000000000';
+
+        describe('simpleSwap', () => {
+          it('MATIC -> TOKEN', async () => {
```

Comment:
I think better to have specific names instead of generic `TOKEN` because it is easy to run particular tests